File: Java/src/main/java/com/DSA/Java/Algorithms/Sorting/SelectionSort.java
package com.DSA.Java.Algorithms.Sorting;

public class SelectionSort {
   public int[] sort(int[] arr) {
      for (int i = 0; i < arr.length; i++) {
         int index = i;
         for (int j = i + 1; j < arr.length; j++) {
            if (arr[j] < arr[index]) {
               index = j;
            }
         }
         
         if (index != i) { 
            int temp = arr[index];
            arr[index] = arr[i];
            arr[i] = temp;
         }
         
      }
      return arr;
   }
}

File: C++/Algorithm/src/Sorting/SelectionSort.cpp
#include "../../include/Sorting/SelectionSort.h"

void SelectionSort::sort_array() {
    for (int i = 0; i < arr.size(); i++) {
        int index = i;
        for (int j = i + 1; j < arr.size(); j++) {
            if (arr[j] < arr[index]) {
                index = j;
            }
        }

        if (index != i) {
            int temp = arr[i];
            arr[i] = arr[index];
            arr[index] = temp;
        }
    }
}
